### PR TITLE
improve builds, step 1: separate e2e tests

### PR DESCRIPTION
**Pull request by pierreca**
# Description of the problem
Build is too slow

# Description of the solution
- Step 1: Separate unit/integration tests from e2e tests (this PR)
- Step 2: Parallelize in separate builds (future PR)

### CODE REVIEW

File: build/build.cmd [hunk 1]
```diff
@@ -125,8 +113,7 @@ echo Lint code and run tests.
 echo build.cmd [options]
 echo options:
 echo  --min                 minimize display output
-echo  --integration-tests   run integration tests too (unit tests always run)
-echo  --e2e-tests           run end-to-end tests too (unit tests always run)
+echo  --integration-tests   run integration tests too (unit tests always run)=
```
Comment (BertKleewein) [inline build/build.cmd]: >= [](start = 77, length = 1)

oops

**Comment by BertKleewein** [comment]
> set IOTHUB_X509_KEY=%node-root%\%IOTHUB_X509_DEVICE_ID%-key.pem

Is this still needed here?  Or only if running integration tests?  Installing OPENSSL is extra burden if unit tests are the only goal -- like when I was working on my laptop last week and was nowhere near running integration/e2e, but did want to run unit tests.

---
Refers to: build/build.cmd:55 in b9cce47. [](commit_id = b9cce47333f901e1b44ba11d7741f83b8f2a00ec, deletion_comment = False)

File: build/e2etests.cmd [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+  echo The OPENSSL_CONF environment variable must be defined in order to generate the x509 certificate for the test device.
+  set ERRORLEVEL=1
+  goto :eof
+)
+
+rem ---------------------------------------------------------------------------
+rem -- lint and run tests
+rem ---------------------------------------------------------------------------
+
+if %e2e-tests%==1 (
```
Comment (BertKleewein) [inline build/e2etests.cmd]: >if %e2e-tests%==1 [](start = 0, length = 17)

copy/paste mistake?

File: build/e2etests.sh [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    echo "options"
+    echo " --min                 minimize display output"
+    exit 1
+}
+
+process_args ()
+{
+    min_output=0
+    integration_tests=0
+    e2e_tests=0
```
Comment (BertKleewein) [inline build/e2etests.sh]: Copy pasta

**Comment by pierreca** [comment]
> set IOTHUB_X509_KEY=%node-root%\%IOTHUB_X509_DEVICE_ID%-key.pem

yeah right now integration and unit tests are kinda stuck together (and integration tests influence test coverage). 

---
In reply to: [339074123](https://github.com/Azure/azure-iot-sdk-node/pull/116#issuecomment-339074123) [](ancestors = 339074123)

---
Refers to: build/build.cmd:55 in b9cce47. [](commit_id = b9cce47333f901e1b44ba11d7741f83b8f2a00ec, deletion_comment = False)

**Pull request merged**